src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,136 @@
+package duke.task;
+
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+
+/**
+ * TaskList store a list of tasks.
+ */
+public class TaskList {
+
+    private ArrayList<Task> taskList;
+
+    /**
+     * Constructs a TaskList object with empty task list.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a TaskList object with the given task list.
+     *
+     * @param taskList Task list that are stored in previous execution.
+     */
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Adds task into the task list.
+     *
+     * @param task Task to be added.
+     */
+    public void addTask(Task task) {
+        this.taskList.add(task);
+    }
+
+    /**
+     * Deletes task in the task list.
+     *
+     * @param taskNumber The task number of the task to be deleted.
+     * @throws DukeException If task number entered is invalid.
+     */
+    public void deleteTask(int taskNumber) throws DukeException {
+        try {
+            this.taskList.remove(taskNumber - 1);
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Please enter a valid task number.");
+        }
+
+    }
+
+    /**
+     * Returns task based on task number.
+     *
+     * @param taskNumber The task number of the task to get.
+     * @return Task of the task number.
+     * @throws DukeException If task number entered is invalid.
+     */
+    public Task getTask(int taskNumber) throws DukeException {
+        try {
+            return this.taskList.get(taskNumber - 1);
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Please enter a valid task number.");
+        }
+
+    }
+
+    /**
+     * Marks task specified as done.
+     *
+     * @param taskNumber The task number of the task to mark as done.
+     * @throws DukeException If task number entered is invalid.
+     */
+    public void doneTask(int taskNumber) throws DukeException {
+        try {
+            this.taskList.get(taskNumber - 1).markDone();
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Please enter a valid task number.");
+        }
+    }
+
+    /**
+     * Returns the number of tasks stored in the task list.
+     *
+     * @return Number of tasks stored in the task list.
+     */
+    public int totalTask() {
```

Comment:
Perhaps a better idea to use a verb (action word) in the method name to improve clarity? How about something like getTotalTask instead of totalTask? This is mentioned in the coding standard guideline, please do take a look at it!